distribution/lib/Standard/Database/0.0.0-dev/src/DB_Column.enso

```diff
@@ -209,6 +209,106 @@ type DB_Column
     to_sql : SQL_Statement
     to_sql self = self.to_table.to_sql
 
+    ## PRIVATE
+       Column-level manual CTE factoring.
+
+       Calling `let` on a column wraps it as a CTE (common table
+       expression), using a SQL `WITH ... AS` clause. More specifically, it
+       takes a callback that receives a "reference" to the CTE; the callback
+       then returns an arbitrary column that uses the reference. The `let`
+       call itself returns the full `WITH` clause, containing both the CTE and
+       the callback return value.
+
+       Using `let` can reduce the number of duplicates of a column expression
+       in the final generated SQL, replacing them with references to a single
+       CTE bound by the `WITH` clause.
+
+       `let` acts like a kind of "let binding". It works by giving a lexically
+       scoped name to the query generated by `self`, and then generating the
+       query returned by the callback inside of this scope. See the examples
+       below to see how the generated SQL is structured.
+
+       Internally, `let` generates a unique name for the CTE, and creates a
+       "reference" column which refers to that unique name. This
+       "reference" column is passed to the callback, which can compute values
+       based on the reference and any other values. Finally, the return value of
+       the callback is wrapped in the binding `WITH ... AS` clause, which is
+       returned from the original call to `let`.
+
+       `let` is only available in database backends that support `WITH` clauses
+       inside expressions. For database backends that only support a single
+       `WITH` claus at the top level, `let` returns `self` unchanged.
+
+       Semantically, the following expressions will always have the same value:
+
+           1. f column
+           2. column.let f
+
+       ? When to use `let`
+
+         `let` can make queries shorter and/or simpler by eliminating
+         duplicates. However, the `WITH` clause itself, including the bound CTE
+         table name, also takes up space, so if the `self` argument to `let`
+         isn't very large, `let` can actually make the query longer.
+
+         For this reason, it is generally better to apply `let` to values
+         before passing them into a library method, rather than to apply it to
+         the value inside the library method itself.
```

Comment:
Tbh I do not understand this paragraph.

After reading the code I understand that you mean that e.g. with `iif` it was not a good idea to use `let` as part of `iif`, but instead if you know that a big expression is being passed as the branching argument to `iif`, you are using `let` at the call-site.

But I couldn't figure it out from this paragraph too easily. I think I'd rewrite it to maybe just say sth like 'For this reason, it makes sense to use `let` only on expressions that at the same time: are expected to be relatively large (results of complex transformations already) and are going to be repeated multiple times in the query.'

Reply:
Done